lib/xhr-mock-proxy-handler.js

```diff
@@ -0,0 +1,50 @@
+var window              = require('global');
+var Promise             = require('lie');
+var realXMLHttpRequest  = window.XMLHttpRequest;
+
+function parseXhrHeaders(xhr) {
+    var headers = {};
+    var rawHeaders = xhr.getAllResponseHeaders().split('\r\n');
+
+    for (var i = 0; i < rawHeaders.length; i++) {
+        var rawHeader = rawHeaders[i].trim();
+                
+        if (rawHeader !== '') {
+            var splittedHeader = rawHeader.split(": ");
+            var headerName = splittedHeader[0];
+            var headerValue = splittedHeader[1];
+            headers[headerName] = headerValue;
+        }
+    }
+
+    return headers;
+}
+
+module.exports = function(req, res) {
+    return new Promise(function(resolve, reject) {
+        var xhr = new realXMLHttpRequest();
```

Comment:
I wasn't unable to use `axios` as you recommended, because it access window.XmlHttpRequest, which can be hidden by xhr-mock. I'm afraid that some else wrapper libraries could do same thing :(

This should be enough portable, maybe some old IE could cause problems. But as I understand you mock XMLHttpRequest as well, not some ActiveX objects from old IEs.

If you mean portability on NodeJS there is xmlhttprequest https://www.npmjs.com/package/xmlhttprequest that behaves like regular XMLHttpRequest.